src/Database/Redis/Core.hs

```diff
@@ -2,7 +2,7 @@
     MultiParamTypeClasses, FunctionalDependencies, FlexibleInstances, CPP #-}
 
 module Database.Redis.Core (
-    Connection, connect,
+    Connection(..), connect,
```

Comment:
This is kept internal to hedis, the `Core` module is never exported so this isn't leaking out anywhere in the public API.  I just needed it in PubSub.hs.  Now if you are talking about protecting Core.hs from the rest of hedis itself I guess we can use a `unConnection` destructor.